src/main/java/net/fabricmc/tinyremapper/ClassInstance.java

```diff
@@ -577,11 +586,35 @@ private MemberInstance getMemberPartial(MemberType type, String idPrefix) {
 		return ret;
 	}
 
+	ClassInstance constructMrjCopy() {
+		// isInput should be false, since the MRJ copy should not be emitted
+		ClassInstance cls = new ClassInstance(context, false, inputTags, mrjVersion, srcPath, data);
+		cls.init(name, superName, access, interfaces);
+		// shadow copy the members and resolvedMembers, since the members should follow the ClassInstance
+		// regardless of the version
+		cls.members = members;
+		cls.resolvedMembers = resolvedMembers;
```

Comment:
I meant to shallow copy the members themselves, which entails a 1-deep copy of the collection. In the case for `members` this entails leaving the members collection as-is (default empty collection), then 
```java
for (MemberInstance oldMember : origCls.getMembers()) {
	newCls.addMember(new MemberInstance(oldMember.type, newCls, oldMember.name, ...);
}
```

This ensures that MembeRInstance.cls is accurate, and the propagation-result captured by newName, etc stays in isolation.

Reply:
fixed